qa/integration-tests/src/test/java/io/camunda/it/client/UserTaskQueryTest.java

```diff
@@ -600,6 +601,427 @@ void shouldReturnUserTaskVariablesWithSubProcessVariables() {
+    final var userTaskListComplete =
+        camundaClient
+            .newUserTaskQuery()
+            .filter(f -> f.state(TaskState.COMPLETED.name()))
+            .send()
+            .join();
+
+    final var userTaskCompletionDateExample =
+        OffsetDateTime.parse(
+            userTaskListComplete.items().stream().findFirst().get().getCompletionDate());
+
+    final var result =
+        camundaClient
+            .newUserTaskQuery()
+            .filter(
+                f ->
+                    f.completionDate(
+                        b ->
+                            b.gte(userTaskCompletionDateExample.minusDays(1))
+                                .lte(userTaskCompletionDateExample.plusDays(1))))
+            .send()
+            .join();
+
+    // then all items have the completion Date within the range
+    assertThat(result.items().size()).isGreaterThan(0);
+    result
+        .items()
+        .forEach(
+            item -> {
+              final var completionDate = OffsetDateTime.parse(item.getCompletionDate());
+              assertThat(completionDate)
+                  .isBetween(
+                      userTaskCompletionDateExample.minusDays(1),
+                      userTaskCompletionDateExample.plusDays(1));
+            });
+  }
+
+  @Test
+  void shouldReturnUserTaskByCompletionDateGtLt() {
+    // when
+    final var userTaskListComplete =
+        camundaClient
+            .newUserTaskQuery()
+            .filter(f -> f.state(TaskState.COMPLETED.name()))
+            .send()
+            .join();
+
+    final var userTaskCompletionDateExample =
+        OffsetDateTime.parse(
+            userTaskListComplete.items().stream().findFirst().get().getCompletionDate());
+
+    final var result =
+        camundaClient
+            .newUserTaskQuery()
+            .filter(
+                f ->
+                    f.completionDate(
+                        b ->
+                            b.gt(userTaskCompletionDateExample.minusDays(1))
+                                .lt(userTaskCompletionDateExample.plusDays(1))))
+            .send()
+            .join();
+
+    // then all items have the completion Date strictly within the range (exclusive)
+    assertThat(result.items().size()).isGreaterThan(0);
+    result
+        .items()
+        .forEach(
+            item -> {
+              final var completionDate = OffsetDateTime.parse(item.getCompletionDate());
+              assertThat(completionDate).isAfter(userTaskCompletionDateExample.minusDays(1));
+              assertThat(completionDate).isBefore(userTaskCompletionDateExample.plusDays(1));
+            });
+  }
+
+  @Test
+  void shouldReturnUserTaskByCompletionDateExists() {
+    // when
+    final var result =
+        camundaClient
+            .newUserTaskQuery()
+            .filter(f -> f.completionDate(b -> b.exists(true)))
+            .send()
+            .join();
+
+    // then
+    assertThat(result.items().size()).isGreaterThanOrEqualTo(0);
```

Comment:
🔧 Let's make sure we have at least one completed user task that we can search for here. Otherwise, we cannot really validate this.